src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -3402,14 +3402,19 @@ theorem and_one_eq_setWidth_ofBool_getLsbD {x : BitVec w} :
 theorem replicate_zero {x : BitVec w} : x.replicate 0 = 0#0 := by
   simp [replicate]
 
+@[simp]
+theorem replicate_one {w : Nat} {x : BitVec w} (h : w = w * 1 := Eq.symm (Nat.mul_one _)) :
```

Comment:
I don't understand why `h` is an argument here. Just inline the default value into the argument of the cast. Proof irrelevance ensures that `simp` and `rw` can still apply the lemma even if a "different" proof appears.